server/py/services/api/api/endpoints/model_monitoring.py

```diff
@@ -310,7 +312,7 @@ async def delete_model_monitoring_function(
 
 
 @router.put("/credentials")
-def set_model_monitoring_credentials(
+async def set_model_monitoring_credentials(
```

Comment:
```suggestion
def set_model_monitoring_credentials(
```

leave as is, fastapi will take care of it for you